ouroboros-network/src/Ouroboros/Network/Mux/Ingress.hs

```diff
@@ -112,16 +116,25 @@ demux pmss = forever $ do
     -- this happens @'Ouroboros.Network.Socket.socketAsMuxBearer'@ throws an
     -- exception.
     (sdu, _) <- Ouroboros.Network.Mux.Types.read $ bearer pmss
-    --say $ printf "demuxing sdu on mid %s mode %s" (show $ msId sdu) (show $ msMode sdu)
-    -- Notice the mode reversal, ModeResponder is delivered to ModeInitiator and vice versa.
-    atomically $ writeTBQueue (ingressQueue (dispatchTable pmss) (msId sdu) (negMiniProtocolMode $ msMode sdu)) (msBlob sdu)
+    -- say $ printf "demuxing sdu on mid %s mode %s lenght %d " (show $ msId sdu) (show $ msMode sdu)
+    --             (BL.length $ msBlob sdu)
+    atomically $ do
+        -- Notice the mode reversal, ModeResponder is delivered to ModeInitiator and vice versa.
+        let q = ingressQueue (dispatchTable pmss) (msId sdu) (negMiniProtocolMode $ msMode sdu)
+        buf <- readTVar q
+        if BL.length buf + BL.length (msBlob sdu) <= maximumIngressQueue (msId sdu)
```

Comment:
I would also document this requirement here so it's visible in haddock: https://github.com/input-output-hk/ouroboros-network/pull/431/files#diff-dae6ca21b7d47354b98fc711cea7eed9L38

Reply:
Will address.